File: ruby/bogosort.rb
def sorted? arr
  arr.each_cons(2).all? { |a, b| a <= b }
end

def shuffle arr
  for i in 0..arr.length()-1
    j = rand(arr.length())
    arr[i], arr[j] = arr[j], arr[i]
  end
end

array = [10, 2, 7, 4, 6, 1, 9, 0]
shuffle array until sorted? array
print array

File: rust/bogosort.rs
extern crate rand;
use rand::{thread_rng, Rng};

fn shuffle(arr: &mut [i32]) {
    let mut rng = thread_rng();
    for i in 0..arr.len() {
        let j = rng.gen_range(0..arr.len());
        (arr[i], arr[j]) = (arr[j], arr[i]);
    }
}

fn is_sorted(arr: &[i32]) -> bool {
    for i in 1..arr.len() {
        if arr[i-1] >= arr[i] {
            return false;
        }
    }
    true
}

fn main() {
    let mut arr: [i32; 8] = [10, 2, 7, 4, 6, 1, 9, 0];
    while !is_sorted(&arr) {
        shuffle(&mut arr);
    }
    println!("{:?}", arr);
}
